toolbox/__init__.py

```diff
@@ -237,22 +237,30 @@ def vocoder_progress(i, seq_len, b_size, gen_rate):
         wav = wav / np.abs(wav).max() * 0.97
         self.ui.play(wav, Synthesizer.sample_rate)
 
-        #Enable replay and save buttons:
-        self.ui.replay_wav_button.setDisabled(False)
-        self.ui.export_wav_button.setDisabled(False)
+        # Name it (history displayed in combobox)
+        # TODO better naming for the combobox items?
+        wav_name = str(self.waves_count + 1)
 
         #Update waves combobox
         self.waves_count += 1
-        if self.waves_count >= MAX_WAVES:
-          self.waves_list.pop(0)          
-        self.waves_list.append(wav)
-        #TODO better naming for the combobox items?
-        self.waves_namelist = ["%d" % (self.waves_count - i) for i in range(0, min(self.waves_count, MAX_WAVES))]
+        if self.waves_count > MAX_WAVES:
+          self.waves_list.pop()
+          self.waves_namelist.pop()
+        self.waves_list.insert(0, wav)
+        self.waves_namelist.insert(0, wav_name)
+
         self.ui.waves_cb.disconnect()
         self.ui.waves_cb_model.setStringList(self.waves_namelist)
         self.ui.waves_cb.setCurrentIndex(0)
         self.ui.waves_cb.currentIndexChanged.connect(self.set_current_wav)
 
+        # Update current wav
+        self.set_current_wav(0)
```

Comment:
For some reason I needed this to ensure that the wav would update itself without combobox interaction. Otherwise it would play a stale wav when clicking "replay" immediately after generating a new one.